Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -1214,86 +1195,58 @@ TextureCacheBase::TCacheEntry* TextureCacheBase::Load(const u32 stage)
 }
 
 TextureCacheBase::TCacheEntry*
-TextureCacheBase::GetTexture(u32 address, u32 width, u32 height, const TextureFormat texformat,
-                             const int textureCacheSafetyColorSampleSize, u32 tlutaddr,
-                             TLUTFormat tlutfmt, bool use_mipmaps, u32 tex_levels, bool from_tmem,
-                             u32 tmem_address_even, u32 tmem_address_odd)
+TextureCacheBase::GetTexture(const int textureCacheSafetyColorSampleSize, TextureInfo& texture_info)
 {
-  // TexelSizeInNibbles(format) * width * height / 16;
-  const unsigned int bsw = TexDecoder_GetBlockWidthInTexels(texformat);
-  const unsigned int bsh = TexDecoder_GetBlockHeightInTexels(texformat);
+  unsigned int expandedWidth = texture_info.GetExpandedWidth();
+  unsigned int expandedHeight = texture_info.GetExpandedHeight();
```

Comment:
This should be `expanded_width`, and possibly also `u32` instead of `unsigned int`.  I also find the logic around this a bit confusing (both that and `width` are updated by the hires texture logic, but in a way I don't find super obvious).

Reply:
Agreed.  TextureCacheBase is very confusing.  This PR isn't mean to completely solve that, just take a step in cleaning things up a bit (though the main purpose is for an enhancement).